theme/theme.inc

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * @file
+ * This file contains all theme and preprocess functions.
+ */
+
+/**
+ * Implements template_preprocess_HOOK().
+ *
+ * @param array $variables
+ *   The theme template variables.
+ */
+function template_preprocess_islandora_openseadragon_viewer(array &$variables) {
+  module_load_include('inc', 'islandora_openseadragon', 'includes/utilities');
+  $library_path = libraries_get_path('openseadragon');
+  if (isset($variables['fedora_object']) && $variables['fedora_object'] !== NULL) {
+    $message = t("The parameter 'fedora_object' in theme('islandora_openseadragon_viewer') has been deprecated. As of 7.x-1.9, please update your code before the next release.");
+    trigger_error(check_plain($message), E_USER_DEPRECATED);
+    // Attempt to convert 'fedora_object', to 'pid'.
+    $variables['pid'] = $variables['fedora_object']->id;
+  }
+  if (isset($variables['uri']) && $variables['uri'] !== '') {
+    $message = t("The parameter 'uri' in theme('islandora_openseadragon_viewer') has been deprecated. As of 7.x-1.9, please update your code before the next release.");
+    trigger_error(check_plain($message), E_USER_DEPRECATED);
+    // Attempt to convert the 'uri' to 'tile_sources'.
+    $matches = array();
+    preg_match('/datastream\/([^\/]*)\/view/', $variables['uri'], $matches);
```

Comment:
Isn't this the same thing that is done on the callback function at https://github.com/Islandora/islandora_openseadragon/pull/65/files#diff-379b938af47456a086e450153e57f124R102?